### ISSUE TITLE
Routing: Jumping across highways

### ISSUE THREAD

**Issue by hybridOL** [mapsme/omim#5410]
I found a strange routing which jumps from one street to another where no connection between those streets exist. As it is a german highway, there's even no way to cross there. I double-checked the OSM data. There's definitely no connection of any kind, and it never existed. This is exit from highway A1 in Osnabrück-Hafen. Maps.Me version 7.1.3-Google, data 170119, Android 6. All maps were updated properly.

![screenshot_20170213-102301](https://cloud.githubusercontent.com/assets/14997736/22880502/b600ea88-f1e2-11e6-99af-72daf1b00773.png)

**Comment by utack** [comment]
Is this on a map border?  
If yes it might be the same as https://github.com/mapsme/omim/issues/3284

**Comment by hybridOL** [comment]
I'm not sure how to find out which parts of the roads are put into which map part. There is a "border" near this point, though, as you can see on the picture. The black line separates Lower Saxony from Northrhine Westfalia. Thus, roads might be split up here.

**Comment by utack** [comment]
Just delete one map.
And voila, it is the same bug
![screenshot_20170214-124847](https://cloud.githubusercontent.com/assets/8551234/22928483/f9469444-f2b6-11e6-951e-62caa7fbcc6c.png)

**Comment by utack** [comment]
Also suspecting this one is similar  
https://github.com/mapsme/omim/issues/4424

**Comment by hybridOL** [comment]
Yes, indeed. For some reason, the streets in Paris are really mixed up in different maps, and that location also crosses map borders.

**Comment by burivuh** [comment]
Hi!

We'll add you example to our regression testing for a 7.3 version where cross-mwm routing will be redone.

Current issues are related to a mixture of an OSRM cross-mwm routing and our custom in-mwm routing. All this will be fixed in a month or so after we'll remove OSRM part completely.

Thank you!
